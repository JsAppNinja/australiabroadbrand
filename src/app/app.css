#app {
  text-align: center;
}

#header {
  background-color: #222;
  height: auto;
  padding: 20px;
  color: white;
}

#title {
  margin-top: 0px;
  font-size: 1.5em;
}

#content {
  margin-top: 20px;
}
