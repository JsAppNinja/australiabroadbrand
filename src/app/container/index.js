import React from "react";
import { Route, Switch } from "react-router-dom";
import AuthenticatedRoute from "../components/authenticated-route";
import UnauthenticatedRoute from "../components/unauthenticated-route";
import Loadable from "react-loadable";

import NotFound from "./not-found";

const Home = Loadable({
  loader: () => import("./home"),
  loading: () => null,
  modules: ["home"]
});

const Login = Loadable({
  loader: () => import("./login"),
  loading: () => null,
  modules: ["login"]
});

const Logout = Loadable({
  loader: () => import("./logout"),
  loading: () => null,
  modules: ["logout"]
});

export default () => (
  <Switch>
    <Route exact path="/" component={Home} />

    <UnauthenticatedRoute exact path="/login" component={Login} />
    <AuthenticatedRoute exact path="/logout" component={Logout} />

    <Route component={NotFound} />
  </Switch>
);
