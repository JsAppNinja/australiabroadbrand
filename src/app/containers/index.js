import React from 'react';
import { Route, Switch } from 'react-router-dom';
//import AuthenticatedRoute from "../components/authenticated-route";
import UnauthenticatedRoute from '../components/unauthenticated-route';
import Loadable from 'react-loadable';

import NotFound from './not-found';

const Home = Loadable({
  loader: () => import('./home'),
  loading: () => null,
  modules: ['home'],
});

const NbnExplained = Loadable({
  loader: () => import('./nbn-explained'),
  loading: () => null,
  modules: ['nbn-explained'],
});

const Login = Loadable({
  loader: () => import('./login'),
  loading: () => null,
  modules: ['login'],
});

const About = Loadable({
  loader: () => import('./about'),
  loading: () => null,
  modules: ['about'],
});

export default () => (
  <Switch>
    <Route exact path="/" component={Home} />
    <Route exact path="/nbn-explained" component={NbnExplained} />
    <Route exact path="/about" component={About} />
    <Route component={NotFound} />

    <UnauthenticatedRoute exact path="/login" component={Login} />
    {/* <AuthenticatedRoute exact path="/logout" component={Logout} /> */}
  </Switch>
);
