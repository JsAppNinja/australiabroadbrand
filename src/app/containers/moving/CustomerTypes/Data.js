import NewCustomer from '../../../assets/new-customer.jpg';
import ExistingCustomer from '../../../assets/existing-customer.jpg';

export default {
  newCustomer: {
    type: 'new-customer',
    text: "I'm a new customer",
    img: NewCustomer,
  },
  existingCustomer: {
    type: 'existing-customer',
    text: "I'm an existing customer",
    img: ExistingCustomer,
  },
};
