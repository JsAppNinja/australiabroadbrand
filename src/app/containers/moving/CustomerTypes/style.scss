.customers {
  display: flex;
  justify-content: center;
  margin-top: 105px;
  background-image: url('../../../assets/customers-background.svg');
  background-repeat: no-repeat;
  background-position-y: center;
  &_backdown {
    margin-bottom: 50px;
  }
}
.customers__card {
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  position: relative;
  height: 295px;
  width: 435px;
  border-radius: 6px;
  box-shadow: 0 8px 20px 0 #e6eaec;
  background-color: #ffffff;
  cursor: pointer;
  &:first-child {
    margin-right: 100px;
  }
  p {
    font-size: 24px;
    letter-spacing: 0;
    font-family: Baloo;
  }
  &.new-customer {
    &.active {
      border: 1px solid #f89729;
      &::after {
        content: '\2713';
        font-size: 15px;
        display: flex;
        justify-content: center;
        align-items: center;
        line-height: 13px;
        font-weight: bold;
        color: white;
        position: absolute;
        top: -10px;
        left: -10px;
        width: 24px;
        height: 24px;
        background-color: #f89729;
        border-radius: 50%;
      }
    }
    p {
      color: #f89729;
    }
  }
  &.existing-customer {
    &.active {
      border: 1px solid #67cebb;
      &::after {
        content: '\2713';
        font-size: 15px;
        display: flex;
        justify-content: center;
        align-items: center;
        line-height: 13px;
        font-weight: bold;
        color: white;
        position: absolute;
        top: -10px;
        left: -10px;
        width: 24px;
        height: 24px;
        background-color: #67cebb;
        border-radius: 50%;
      }
    }
    p {
      color: #67cebb;
    }
  }
  img {
    height: 155px;
    width: 265px;
  }
}
