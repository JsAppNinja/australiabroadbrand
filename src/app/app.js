import React, { Component } from "react";
import { connect } from "react-redux";
import { bindActionCreators } from "redux";
import { withRouter } from "react-router";

import { isServer } from "../store";
import Header from "./containers/home/header";
import Routes from "./routes";

import "./app.css";

class App extends Component {
  componentWillMount() {
    if (!isServer) {
      //  this.props.establishCurrentUser();
    }
  }

  render() {
    return (
      <div id="app">
        <Header
          isAuthenticated={this.props.isAuthenticated}
          current={this.props.location.pathname}
        />
        <div id="content">
          <Routes />
        </div>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  isAuthenticated: state.auth.isAuthenticated
});

export default withRouter(connect(mapStateToProps)(App));
