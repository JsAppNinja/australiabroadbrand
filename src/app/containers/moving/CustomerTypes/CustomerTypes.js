import React, { Component } from 'react';
import Data from './Data';
import PropTypes from 'prop-types';

import './style.scss';

class CustomerTypes extends Component {
  state = {
    selectedCustomer: {
      type: '',
      isActive: false,
    },
  };
  handleClick = (type, isActive) => {
    const { selectedCustomer } = this.state;
    this.setState(
      {
        selectedCustomer: {
          isActive: !selectedCustomer.isActive,
          type: selectedCustomer.type === type ? '' : type,
        },
      },
      () => {
        this.props.toggleSteps(this.state.selectedCustomer);
      }
    );
  };

  render() {
    const { isActive, type } = this.state.selectedCustomer;
    const { newCustomer, existingCustomer } = Data;
    const newCustomerMatch = isActive && type === 'new-customer';
    const existingCustomerFavList = isActive && type === 'existing-customer';
    return (
      <section className={`customers ${!isActive ? 'customers_backdown' : ''}`}>
        <article
          className={`customers__card new-customer ${
            newCustomerMatch ? 'active' : ''
          }`}
          onClick={() => this.handleClick(newCustomer.type, isActive)}
        >
          <img src={newCustomer.img} alt="new-customer" />
          <p>{newCustomer.text}</p>
        </article>
        <article
          className={`customers__card existing-customer ${
            existingCustomerFavList ? 'active' : ''
          }`}
          onClick={() => this.handleClick(existingCustomer.type)}
        >
          <img src={existingCustomer.img} alt="existing-customer" />
          <p className="">{existingCustomer.text}</p>
        </article>
      </section>
    );
  }
}

CustomerTypes.propTypes = {
  toggleSteps: PropTypes.func.isRequired,
};

export default CustomerTypes;
