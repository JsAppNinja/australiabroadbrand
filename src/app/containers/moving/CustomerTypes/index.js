import CustomerTypes from './CustomerTypes';

export default CustomerTypes;
