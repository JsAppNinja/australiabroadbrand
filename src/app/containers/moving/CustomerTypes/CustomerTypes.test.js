import React from 'react';
import { shallow, mount, request } from 'enzyme';

import app from '../../../app';
import CustomerTypes from './CustomerTypes';

const defaultProps = {
  toggleSteps: jest.fn(),
  activeType: null,
};

describe('Customer Types', () => {
  const tree = shallow(<CustomerTypes {...defaultProps} />);
  const instance = tree.instance();
  const spyOnToggleSteps = jest.spyOn(defaultProps, 'toggleSteps');
  instance.forceUpdate();

  it('should render customer types section', () => {
    expect(tree).toMatchSnapshot();
  });

  it('should handle click on new customer', () => {
    const btn = tree.find('.customers__card.new-customer');
    btn.simulate('click');
    expect(spyOnToggleSteps).toHaveBeenCalled();
  });

  it('should handle click on existing customer', () => {
    const btn = tree.find('.customers__card.existing-customer');
    btn.simulate('click');
    expect(spyOnToggleSteps).toHaveBeenCalled();
  });

  it('should mount', () => {
    expect(
      mount(<CustomerTypes {...defaultProps} />).find('.customers')
    ).toHaveLength(1);
  });
});
