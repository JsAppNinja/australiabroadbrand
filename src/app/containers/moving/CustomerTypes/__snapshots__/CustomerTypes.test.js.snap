// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Customer Types should render customer types section 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CustomerTypes
    activeType={null}
    toggleSteps={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <article
          className="customers__card new-customer "
          onClick={[Function]}
        >
          <img
            alt="new-customer"
            src="new-customer.jpg"
          />
          <p>
            I'm a new customer
          </p>
        </article>,
        <article
          className="customers__card existing-customer "
          onClick={[Function]}
        >
          <img
            alt="existing-customer"
            src="existing-customer.jpg"
          />
          <p
            className=""
          >
            I'm an existing customer
          </p>
        </article>,
      ],
      "className": "customers ",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <img
              alt="new-customer"
              src="new-customer.jpg"
            />,
            <p>
              I'm a new customer
            </p>,
          ],
          "className": "customers__card new-customer ",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "new-customer",
              "src": "new-customer.jpg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "I'm a new customer",
            },
            "ref": null,
            "rendered": "I'm a new customer",
            "type": "p",
          },
        ],
        "type": "article",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <img
              alt="existing-customer"
              src="existing-customer.jpg"
            />,
            <p
              className=""
            >
              I'm an existing customer
            </p>,
          ],
          "className": "customers__card existing-customer ",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "existing-customer",
              "src": "existing-customer.jpg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "I'm an existing customer",
              "className": "",
            },
            "ref": null,
            "rendered": "I'm an existing customer",
            "type": "p",
          },
        ],
        "type": "article",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <article
            className="customers__card new-customer "
            onClick={[Function]}
          >
            <img
              alt="new-customer"
              src="new-customer.jpg"
            />
            <p>
              I'm a new customer
            </p>
          </article>,
          <article
            className="customers__card existing-customer "
            onClick={[Function]}
          >
            <img
              alt="existing-customer"
              src="existing-customer.jpg"
            />
            <p
              className=""
            >
              I'm an existing customer
            </p>
          </article>,
        ],
        "className": "customers ",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <img
                alt="new-customer"
                src="new-customer.jpg"
              />,
              <p>
                I'm a new customer
              </p>,
            ],
            "className": "customers__card new-customer ",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "new-customer",
                "src": "new-customer.jpg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "I'm a new customer",
              },
              "ref": null,
              "rendered": "I'm a new customer",
              "type": "p",
            },
          ],
          "type": "article",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <img
                alt="existing-customer"
                src="existing-customer.jpg"
              />,
              <p
                className=""
              >
                I'm an existing customer
              </p>,
            ],
            "className": "customers__card existing-customer ",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "existing-customer",
                "src": "existing-customer.jpg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "I'm an existing customer",
                "className": "",
              },
              "ref": null,
              "rendered": "I'm an existing customer",
              "type": "p",
            },
          ],
          "type": "article",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
